src/osd/osd_types.cc

```diff
@@ -5096,20 +5096,6 @@ void ScrubMap::decode(bufferlist::iterator& bl, int64_t pool)
   ::decode(valid_through, bl);
   ::decode(incr_since, bl);
   DECODE_FINISH(bl);
-
-  // handle hobject_t upgrade
-  if (struct_v < 3) {
-    map<hobject_t, object, hobject_t::BitwiseComparator> tmp;
-    tmp.swap(objects);
```

Comment:
@athanatos collection_list() claims to not return the "end" ghobject_t argument.   It is used at the primary and on the replicas from objects_list_partial() and objects_list_range(), respectively.